app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,25 @@
+module SessionsHelper
+
+  def logged_in?
+    !!session[:user_id]
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
+  end
+
+  def current_user_name
+    session[:user_name]
+  end
+
+  def login(user)
+    session[:user_id] = user.id
+    session[:user_name] = user.username
```

Comment:
I usually only store the id in session. It looks like you want to cache the username. The sessions is generally for persisting state, not for caching.